scripts/variantreport_p_sample.rmd

```diff
@@ -136,6 +133,53 @@ get_protein_mut <- function (vepfile){ # parse together from vep output "Protein
   
   return(locations)
 }
+
+deconv <- function (bulk,sig){
+  
+  rlm_model = suppressWarnings(MASS::rlm(sig,bulk, maxit = 100))
+      
+      
+  rlm_coefficients = rlm_model$coefficients
+  
+  rlm_coefficients = ifelse(rlm_coefficients < 0, 0, rlm_coefficients)
+  
+  sumOfCof = sum(rlm_coefficients)
+  
+  rlm_coefficients = rlm_coefficients / sumOfCof  #normalize so coefficients add to 1
+  
+  as.vector(rlm_coefficients)
+}
+
+getIdentCols <- function (dataframe) {
+    '* This function is for collapsing the column names of identical columns and saving them as a new column names.
+       I had the difficulty that with all the existing functions I would always would have missed the "parent" of the 
+       following duplicates. This workaround here should be definitly improved with a more R-like version, if possible. *'
```

Comment:
Could you please show an example of what this procedure should accomplish?  I'd like to see an input data frame and what it looks like when it comes out of this procedure.

Even better: executable assumptions, i.e. inputs and outputs that are supposed to be equal after applying getIdentCols.

Reply:
What do you mean by "show an example"? Creating something unit test (at least that's how it sound like with what you describe with "even better:...") Or do you want a small rmd with containing all those elements?

Reply:
Well, I don't easily follow the code.  I don't understand what it attempts to accomplish and so I can't judge if it really does what it aims to do.

It would be great to see an actual input to this procedure.  A full data frame and what you expect it to look like in the end.

Reply:
ups, see new comment below.